src/perf/event_provider.cpp

```diff
@@ -469,6 +472,25 @@ const EventDescription sysfs_read_event(const std::string& ev_desc)
     }
     EventDescription event(ev_desc, static_cast<perf_type_id>(type), 0, 0);
 
+    std::ifstream cpus_stream(pmu_path / "cpus");
+    std::string cpus_string;
+    cpus_stream >> cpus_string;
+
+    if (cpus_stream)
+    {
+        const std::regex cpus_regex("([0-9]+)-([0-9]+)");
```

Comment:
I think we need a `parse_int_list` method, as this should technically also be used in the `Topology`.
It should be fine with input like `1,3,5-7` to expand to `int[1,3,5,6,7]`.

Reply:
Wait, that even exists: `lo2s::detail::parse_list` in `topology.hpp`